### DISCUSSION TITLE
Enhance touchpad button remapping to support custom combinations

### DISCUSSION THREAD

**Discussion by Paul-Eau** [YaLTeR/niri#813] [Ideas]
I would like to request an enhancement to the touchpad button remapping feature.

**Current Situation:**
Currently, the `tap-button-map` setting allows for configuring individual buttons, but it does not support remapping to custom combinations like `left-left-left`. Currently it only support `left-middle-right` and `right-middle-left` 

**Requested Feature:**
Extend the `tap-button-map` configuration to support custom button combinations. This would allow users to define mappings as their preferences.

**Use Case:**
This feature would be useful for users who want to use all of the touchpad as a left button click.

Thank you for considering this feature request. Your work on Niri is highly appreciated!

**Comment by YaLTeR** [comment]
Hey, this is a libinput option and it only accepts those two values: https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga7c7637385eaf0f1f29a812c3e48fdc3f

I'm not sure if it's possible/easy to properly this in niri. Might be worth bringing it up with libinput.

**Comment by Paul-Eau** [comment]
Hey, thanks for the info, I guess this should be modifiable in the libinput source code but never mind I figured out that my trackpad was using kernel parameter `psmouse.proto=exps` (I had to do that for a better trackpoint precision), Then I switch back to `auto` (default value) and now the whole trackpad works as a single left click button. I guess this is because the ThinkPad x270 has physical buttons for the trackpoint.
